tawkto/tawkto.php

```diff
@@ -613,6 +603,19 @@ class TawkTo {
 		public function __construct() {
 			$tawkto_settings = new TawkTo_Settings();
 			add_shortcode( 'tawkto', array( $this, 'shortcode_print_embed_code' ) );
+
+			add_action( 'init', array( $this, 'start_session' ) );
```

Comment:
Would suggest moving this to actual hashing and when hashing use the following logic

- If no visitor recognition enabled - no hash generated, no session started
- Check if session exists. If not - start it
- Check if hash exists in session. If so - check config version and email for match. If match - use hash from session
- Hash email and store in session

This will make the whole process more linear and remove the session management overhead for users that do not have secure mode enabled